### ISSUE TITLE
storm wipes ssh config if error is thrown

### ISSUE THREAD

**Issue by grayaii** [emre/storm#111]
I have an ssh file configured like so:
```
======= contents of ~/.ssh/config =========
ServerAliveCountMax 6
ControlMaster auto
ControlPath ~/.ssh/ssh_control_%h_%p_%r
StrictHostKeyChecking no
UserKnownHostsFile=/dev/null
User            alex.gray
IdentityFile    ~/Documents/my_id_rsa

Host github.com
    controlmaster no
====================================
```
And when I run the following command, it completely deletes my ssh config:
storm add my_vps root@emreyilmaz.me:22
```
Traceback (most recent call last):
  File "/Users/alexgray/.virtualenvs/skippy/bin/storm", line 11, in <module>
    sys.exit(main())
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/kommandr.py", line 195, in __call__
    self.execute(sys.argv[1:])
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/kommandr.py", line 188, in execute
    return command(**arg_map)
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/__main__.py", line 51, in add
    storm_.add_entry(name, host, user, port, id_file, o)
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/__init__.py", line 37, in add_entry
    self.ssh_config.write_to_ssh_config()
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/parsers/ssh_config_parser.py", line 228, in write_to_ssh_config
    data = self.dump()
  File "/Users/alexgray/.virtualenvs/skippy/lib/python3.4/site-packages/storm/parsers/ssh_config_parser.py", line 203, in dump
    self.config_data = sorted(self.config_data, key=itemgetter("order"))
TypeError: unorderable types: NoneType() < int()
```
And now my ssh config is completely deleted :(

Storm should do one or all of the following:
1) never wipe the ssh config on a storm failure.
2) possibly create a backup of the ssh config to see what changed.
3) when reading the ssh config, strip everything outside a hosts section, modify the file in memory, and add the text that we stripped BACK into the generated ssh config.

This is using the following version of storm:
$ pip list | grep storm
stormssh (0.6.7)

**Comment by emre** [comment]
Really sorry for the trouble. Can't reproduce the problem though. :(

Can you write your

* exact python version
* OS X version
* install way (pip, easy_install or something else?)
* your sshconfig file (don't copy/paste it. remove sensitive data (if any) andend to hello[at]emre.sh)

**Comment by grayaii** [comment]
No worries! (I update my original comment with the exact contents of my ssh config.)
Here is some information
```
Python 3.4.3
OS: MAC El Capitan 10.11.1 (15B42)
It was installed via pip (pip install stormssh) version: (0.6.7)
```
Hope this helps!

**Comment by emre** [comment]
Fixed w/ https://github.com/emre/storm/commit/8c03351e6613c2298fe806d72097c3899fd3d057.

Can you update and try again? 😊

**Comment by grayaii** [comment]
Works like a charm!
Thanks man!!!
Alex

On Fri, Nov 6, 2015 at 1:39 PM, Emre Yılmaz <notifications@github.com>
wrote:

> Fixed w/ 8c03351
> <https://github.com/emre/storm/commit/8c03351e6613c2298fe806d72097c3899fd3d057>
> .
>
> Can you update and try again? 😊
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/emre/storm/issues/111#issuecomment-154496812>.
>



-- 
-Alex-

**Comment by emre** [comment]
happy to hear, have fun!
